### PR TITLE
TST: Fix for doctest in style_render.py -> pandas.io.formats.style_re…

### LINKED ISSUE #42766: TST: Fix a doctest in style_render.py
Fix the following doctest in style_render.py file:
```
pandas/io/formats/style_render.py F

=================================== FAILURES ===================================
________ [doctest] pandas.io.formats.style_render.StylerRenderer.format ________
647 
648         >>> df = pd.DataFrame([[np.nan, 1.0, 'A'], [2.0, np.nan, 3.0]])
649         >>> df.style.format(na_rep='MISS', precision=3) # doctest: +SKIP
650                 0       1       2
651         0    MISS   1.000       A
652         1   2.000    MISS   3.000
653 
654         Using a ``formatter`` specification on consistent column dtypes
655 
656         >>> df.style.format('{:.2f}', na_rep='MISS', subset=[0,1])
Expected:
            0      1          2
    0    MISS   1.00          A
    1    2.00   MISS   3.000000
Got:
    <pandas.io.formats.style.Styler object at 0x7f77b214ea30>

/home/aneesh/Desktop/Side Projects/ops/pandas/pandas/io/formats/style_render.py:656: DocTestFailure
- generated xml file: /home/aneesh/Desktop/Side Projects/ops/pandas/test-data.xml -
============================= slowest 30 durations =============================
1.14s call     pandas/io/formats/style_render.py::pandas.io.formats.style_render.StylerRenderer.format

(2 durations < 0.005s hidden.  Use -vv to show these durations.)
=========================== short test summary info ============================
FAILED pandas/io/formats/style_render.py::pandas.io.formats.style_render.StylerRenderer.format
============================== 1 failed in 1.70s ===============================
```

---

## OBSERVATION

The mentioned test returns a Styler object which will render properly in (for instance, a notebook), but will fail the doctest as it will simply return the object. 

## SOLUTION

Add images of output rendered, in the doctest wherever required. (The approach is mentioned https://github.com/pandas-dev/pandas/issues/42674#issuecomment-887075744, for the documentation of [background_gradient()](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.background_gradient.html)) or add a ```# doctest: +SKIP``` if necessary.

### PR DESCRIPTION
**Pull request by aneesh98**
This pull request fixes the doctest pandas.io.formats.style_render.StylerRenderer.format

- [X] closes #42766 

Following this commit, all the doctests in style_render.py file passed.

Output:
```
python -m pytest --doctest-modules pandas/io/formats/style_render.py
========================================================= test session starts =========================================================
platform linux -- Python 3.8.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/aneesh/Desktop/Side Projects/ops/pandas, configfile: pyproject.toml
plugins: forked-1.3.0, asyncio-0.15.1, anyio-3.3.0, cov-2.12.1, instafail-0.4.2, hypothesis-6.14.3, xdist-2.3.0
collected 3 items                                                                                                                     

pandas/io/formats/style_render.py ...

--------------------------- generated xml file: /home/aneesh/Desktop/Side Projects/ops/pandas/test-data.xml ---------------------------
======================================================== slowest 30 durations =========================================================
1.06s call     pandas/io/formats/style_render.py::pandas.io.formats.style_render.StylerRenderer.format

(8 durations < 0.005s hidden.  Use -vv to show these durations.)
========================================================== 3 passed in 1.15s ==========================================================

```

### CODE REVIEW

**Comment by attack68** [comment]
is it better to have two spaces before `#` as per PEP?

File: pandas/io/formats/style_render.py [hunk 1]
```diff
@@ -698,7 +700,7 @@ def format(
         Using a ``formatter`` with LaTeX ``escape``.
 
         >>> df = pd.DataFrame([["123"], ["~ ^"], ["$%#"]])
-        >>> s = df.style.format("\\textbf{{{}}}", escape="latex").to_latex()
+        >>> s = df.style.format("\\textbf{{{}}}", escape="latex").to_latex() # doctest: +SKIP
```
Comment (attack68) [inline pandas/io/formats/style_render.py:703]: `to_latex()` actually produces a string result.. maybe this could be doctested.

**Review by attack68 — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style_render.py (hunk 1 above)
Comment (aneesh98) [reply pandas/io/formats/style_render.py:703]: The command line output I got by running this test is not pretty printed actually as it is in the docs: 
```
________ [doctest] pandas.io.formats.style_render.StylerRenderer.format ________
694         ...
695         <td .. ><a href="a.com/&lt;div&gt;&lt;/div&gt;">&lt;div&gt;&lt;/div&gt;</a></td>
696         <td .. ><a href="a.com/&#34;A&amp;B&#34;">&#34;A&amp;B&#34;</a></td>
697         <td .. >NA</td>
698         ...
699 
700         Using a ``formatter`` with LaTeX ``escape``.
701 
702         >>> df = pd.DataFrame([["123"], ["~ ^"], ["$%#"]])
703         >>> df.style.format("\\textbf{{{}}}", escape="latex").to_latex()
Expected:
    \begin{tabular}{ll}
    {} & {0} \\
    0 & \textbf{123} \\
    1 & \textbf{\textasciitilde \space \textasciicircum } \\
    2 & \textbf{\$\%\#} \\
    \end{tabular}
Got:
    '\\begin{tabular}{ll}\n{} & {0} \\\\\n0 & \\textbf{123} \\\\\n1 & \\textbf{\\textasciitilde \\space \\textasciicircum } \\\\\n2 & \\textbf{\\$\\%\\#} \\\\\n\\end{tabular}\n'

```
Please let me know if I am missing something here, or have done something incorrectly. 
Thanks.

**Comment by aneesh98** [comment]
> is it better to have two spaces before `#` as per PEP?

Sorry actually I have added two spaces before `#` , referring to other doctests in other files (for instance frame.py). Is it incorrect?

**Review by attack68 — APPROVED** [review]
lgtm, all of these can be skipped

**Comment by aneesh98** [comment]
> lgtm, all of these can be skipped

Thanks

**Comment by KrishnaSai2020** [comment]
@aneesh98 I would recommend merging master into your branch. #42790 and #42820 fixes the CI checks so they should now be all green once you merge

**Comment by aneesh98** [comment]
> @aneesh98 I would recommend merging master into your branch. #42790 and #42820 fixes the CI checks so they should now be all green once you merge

Hi @KrishnaSai2020. Thank you for this.

**Comment by attack68** [comment]
thanks @aneesh98

**Pull request merged**